homeassistant/components/motion_blinds/config_flow.py

```diff
@@ -31,33 +37,71 @@ def __init__(self):
         """Initialize the Motion Blinds flow."""
         self.host = None
         self.key = None
+        self.ips = []
 
     async def async_step_user(self, user_input=None):
         """Handle a flow initialized by the user."""
         errors = {}
         if user_input is not None:
-            self.host = user_input[CONF_HOST]
-            self.key = user_input[CONF_API_KEY]
-            return await self.async_step_connect()
+            self.host = user_input.get(CONF_HOST)
+
+            if self.host is not None:
+                return await self.async_step_connect()
+
+            # Use MotionGateway discovery
+            discover_class = MotionDiscovery()
+            gateways = await self.hass.async_add_executor_job(discover_class.discover)
+            self.ips = list(gateways.keys())
```

Comment:
keys can be omitted:

```suggestion
            self.ips = list(gateways)
```

Reply:
> 
> 
> Very nice.
> 
> Optional: Did you consider flipping this around at all? That is, doing auto discovery on the first step, then falling back to a manual entry if now devices are found within the 10 second timeout? Wasn't sure if leaning even more into the multicast flow is a good idea, or how hard it is from a config flow point of view.
> 
> (I'm looking forward to redoing my network a bit to be able to use this, since multicast seems nice)

Thanks!

The disadvantage of flipping it around will be that if a user selects the motion_blinds from the list of integrations, it will first "hang" for 10 seconds withouth the user knowing what is going on and probably going to be inpatient.

Therefore I think it is better to imediatly prompt the user with a dialog (in the title it is then explained they can opt for discovery instead of manually filling in the IP).